http_2.rflx

```diff
@@ -0,0 +1,230 @@
+package HTTP_2 is
+
+	type Payload_Length is mod 2**24; -- Must not be greater than 2**14 unless the receiver has set a larger value for SETTINGS_MAX_FRAME_SIZE
```

Comment:
Is the maximum frame size negotiated for all future frames? In this case, that needs to be represented in session specification.

Reply:
2^14 is the initial frame size. A larger frame size can be set by end endpoint by sending a Settings Frame with the `SETTINGS_MAX_FRAME_SIZE` parameter set and a value between 2^14 and 2^24-1. If a value outside of this range must be treated as a `PROTOCOL_ERROR`.

Added a constraint to ensure the minimum size of 2^14 in 9bd636f

Reply:
> Added a constraint to ensure the minimum size of 2^14 in 9bd636f

A constraint cannot be added as some frame types need a fixed smaller value e.g. `WINDOW_UPDATE`.
Commit has been reverted: f6b8a94

Reply:
Could this be added as a constraint for the respective frame types in the model? Something like `Frame_Type = Data and Payload_Length >= 2**14 or ...` between `Frame_Type` and `Flag_End_Stream_ACK`?

Reply:
From my understanding of the RFC `SETTINGS_MAX_FRAME_SIZE` allows indicating to a remote station that larger frames can be received:

>    SETTINGS_MAX_FRAME_SIZE (0x5):  Indicates the size of the largest
      frame payload that the sender is willing to receive, in octets.

That cannot be enforced in the message specification, but only in the session specification.

Reply:
> That cannot be enforced in the message specification, but only in the session specification.

Yes, we should leave it for now as it is.